In [139]:
import pandas as pd
import numpy as np

### 1.1

In [140]:
sc = pd.read_csv("./covid-scandinavia.csv.bz2")

I found the columns combined into on single column.

### 1.2

In [141]:
sc.shape

(6528, 1)

### 1.3

In [142]:
sc = pd.read_csv("./covid-scandinavia.csv.bz2", sep = "\t")

### 1.4

In [143]:
sc.shape

(6528, 11)

It has 6528 rows and 11 columns

### 2.1

In [144]:
sc.dtypes

code2          object
country        object
state         float64
date           object
type           object
count           int64
lockdown       object
population      int64
countPC       float64
growth        float64
growthPC      float64
dtype: object

### 2.2

In [145]:
sc = sc[["country", "date", "type", "count", "population"]].copy()
print(sc.shape)
sc.head(5)

(6528, 5)


country        date       type  count  population
0  Denmark  2020-01-22  Confirmed      0     5837213
1  Denmark  2020-01-22     Deaths      0     5837213
2  Denmark  2020-01-23  Confirmed      0     5837213
3  Denmark  2020-01-23     Deaths      0     5837213
4  Denmark  2020-01-24  Confirmed      0     5837213

#### 2.3.a

In [146]:
sc.country.value_counts()

country
Denmark    1632
Finland    1632
Norway     1632
Sweden     1632
Name: count, dtype: int64

They seems to be correctly labelled and no strange values.

#### 2.3.b

In [147]:
print("Max date: ", sc.date.max())
print("Min date: ", sc.date.min())

Max date:  2022-04-16
Min date:  2020-01-22


The results are realistic since covid starts from 2020 from description. The data ends at 2022 which is also reasonable as covid spread for roughly two years.

#### 2.3.c

In [148]:
print("Max Count: ", sc["count"].max())
print("Min Count: ", sc["count"].min())

Max Count:  3097015
Min Count:  0


I would say it is reasonable since there should be somewhere that not affected by covid at the really beginning, which is 0 cases. Covid is also affecting tons of people so the max 3097015 is reasonable for large cities and such.

That is maybe because .count is a built in method in python, so it would not be recognize as series or data frame.

In [149]:
sc.head(10)

country        date       type  count  population
0  Denmark  2020-01-22  Confirmed      0     5837213
1  Denmark  2020-01-22     Deaths      0     5837213
2  Denmark  2020-01-23  Confirmed      0     5837213
3  Denmark  2020-01-23     Deaths      0     5837213
4  Denmark  2020-01-24  Confirmed      0     5837213
5  Denmark  2020-01-24     Deaths      0     5837213
6  Denmark  2020-01-25  Confirmed      0     5837213
7  Denmark  2020-01-25     Deaths      0     5837213
8  Denmark  2020-01-26  Confirmed      0     5837213
9  Denmark  2020-01-26     Deaths      0     5837213

Each date, each country would have 2 rows, one confirmed covid case and death covid case. One row is either representing number of deaths or number of confirmed cases in a certain country in a certain date with population as reference. 

### 3.1 

I'm interested in the general idea of machine learning and I wish to explore more about different models here.

### 3.2

I want to learn how and when to use different models and how to tune them for better performances. 

### 4.1

In [150]:
sc = sc[sc["type"] == "Confirmed"]
sc.head()

country        date       type  count  population
0  Denmark  2020-01-22  Confirmed      0     5837213
2  Denmark  2020-01-23  Confirmed      0     5837213
4  Denmark  2020-01-24  Confirmed      0     5837213
6  Denmark  2020-01-25  Confirmed      0     5837213
8  Denmark  2020-01-26  Confirmed      0     5837213

### 4.2

In [151]:
sc["ccpc"] = sc["count"] / sc["population"]
sc.head()

country        date       type  count  population  ccpc
0  Denmark  2020-01-22  Confirmed      0     5837213   0.0
2  Denmark  2020-01-23  Confirmed      0     5837213   0.0
4  Denmark  2020-01-24  Confirmed      0     5837213   0.0
6  Denmark  2020-01-25  Confirmed      0     5837213   0.0
8  Denmark  2020-01-26  Confirmed      0     5837213   0.0

### 4.3

I didn't get that warning.

### 4.4

In [152]:
swedenCpc = sc[sc["country"] == "Sweden"][["ccpc"]]

### 4.5

In [153]:
type(swedenCpc)

pandas.core.frame.DataFrame

### 4.6

In [154]:
swedenCpc.sample(5)

ccpc
5422  9.486710e-03
5682  6.041966e-02
5776  8.045969e-02
4948  9.635971e-08
5314  7.898798e-03

### 4.7

I saw the cases per capita I created on the right and some index on the left. While the index doesn't really make sence to me, is seems like the index generated by ```pd.read_csv```. A more meaningful index should give me some information about time at least.

### 4.8

In [155]:
swedenCpc1 = sc[sc["country"] == "Sweden"].set_index("date")[["ccpc"]]
swedenCpc1.sample(5)


ccpc
date                
2020-11-23  0.020071
2021-08-01  0.106000
2020-06-17  0.005318
2022-02-27  0.235228
2021-02-16  0.059538